lib/ansible/module_utils/ansible_tower.py

```diff
@@ -45,31 +47,38 @@ def tower_auth_config(module):
     it will attempt to fetch values from the module pararms and
     only pass those values that have been set.
     '''
+    auth_config = {}
+
+    settings = Settings()
+    settings._parser_names = ['ansible', 'environment', 'local', 'user', 'global', 'defaults']
+    settings._ansible = settings._new_parser()  # a new parser seems better than reusing 'runtime' parser
```

Comment:
> # a new parser seems better than reusing 'runtime' parser

I don't know that I agree, because even after you do this processing and return `auth_config`, those values get applied as runtime settings in the `settings` initialized inside of the tower-cli imports. It seems like with what you're doing, file settings will also get picked up by your settings instance, precedence order will be handled with this object, and then all of that will applied as runtime settings. Maybe even defaults, I would need to test.

Reply:
I miss the fact Ansible Tower modules use `settings.runtime_values`, thanks for pointing that.